### ISSUE TITLE
テーブル設計 #1

### ISSUE THREAD

**Issue by yohei04** [yohei04/techpit_tweet_app#1]
## Issue番号
#1

## 背景
テーブル設計設計を行いました。
「ER図」でググったところテーブル名にはsをつけてない物が多かったので単数形表記にしてます。

## 確認項目
+ ER図がちゃんと書けている

**Comment by daijiromaeyama** [comment]
テーブル設計お疲れさまです！気になったところを以下にコメントしていきますね。

## 全体
- Railsの場合、テーブル名は複数形なので複数形にした方がいいかなと思いました
- 作成・更新日時は必要ではないでしょうか
- 主キー・外部キーが記載されていないので記載していただければと思います

## relationshipsテーブル
今回作成するアプリケーションでユーザー数が膨大になることはないと思いますが、Rails5.1からidカラムはデフォルトで`bigint`になっていることを踏まえて、`bigint`にした方がいいかなと思いました。他のテーブルのidがつくカラムもご確認お願いします。

## tweetsテーブル
1つのツイートに対して画像を複数投稿する際はどのようにデータを格納するのでしょうか。

以上です。
ご確認のほどよろしくお願いいたします。

**Comment by yohei04** [comment]
+ Railsの場合、テーブル名は複数形なので複数形にした方がいいかなと思いました
+ 作成・更新日時は必要ではないでしょうか
+ 主キー・外部キーが記載されていないので記載していただければと思います
+ bigintにした方がいい

→　上記4点全て修正致しました。

+ 1つのツイートに対して画像を複数投稿する際はどのようにデータを格納するのでしょうか。

→　[この記事](https://qiita.com/ggtmtmgg/items/ba5f275c122c83013ea1) よりCarrierWaveというGemを使って実現できそう。
　　tweetsテーブルのカラム名を複数形(images)に変えてます。


<img width="708" alt="スクリーンショット 2020-02-05 1 31 14" src="https://user-images.githubusercontent.com/54460011/73765076-61958e80-47b7-11ea-8b91-8193cf6c9367.png">

**Comment by daijiromaeyama** [comment]
修正ありがとうございます！

ツイートの画像以外は大丈夫そうですね。
ツイート画像に関してですが、

> 1つのツイートに対して画像を複数投稿する際は

これは `ツイート : 画像 = 1 : 多`

の関係になりうるので、ツイートの画像用のテーブルを作成するといいかなと思いました。
ご確認のほどよろしくお願いいたします。

**Comment by yohei04** [comment]
レビューありがとうございます！
再度添付致しました。
ご確認お願い致します。

<img width="609" alt="スクリーンショット 2020-02-05 22 19 36" src="https://user-images.githubusercontent.com/54460011/73845361-c90a1780-4865-11ea-9d36-673158fa4e2e.png">

**Comment by daijiromaeyama** [comment]
tweet_picturesテーブルの追加ありがとうございます。気になった箇所を下記に記載しますね。

まず、tweetsテーブルの`image`というカラムはどういったデータを保存するカラムでしょうか。
またtweet_picturesの`image_id` というカラムは外部キーとなっていますがどのテーブルの主キーに値するのでしょうか。

またtweet_pictruesテーブルに画像を保存するカラムがないかなと思いました。

ご確認のほどよろしくお願いいたします。

**Comment by yohei04** [comment]
>tweetsテーブルのimageというカラムはどういったデータを保存するカラムでしょうか。

画像を保存するものだと思っていましたがテーブルがもうひとつあるということはいらないですかね？


>tweet_picturesのimage_id というカラムは外部キーとなっていますがどのテーブルの主キーに値するのでしょうか。

tweetですね。。変更しました。

>tweet_pictruesテーブルに画像を保存するカラムがない

imageカラムを新しく作りました。

<img width="609" alt="スクリーンショット 2020-02-06 9 20 59" src="https://user-images.githubusercontent.com/54460011/73895182-09977e80-48c2-11ea-96f3-b16079e76ecc.png">

**Comment by daijiromaeyama** [comment]
> 画像を保存するものだと思っていましたがテーブルがもうひとつあるということはいらないですかね？

そうですね。新しく追加していただいたtweet_picturesに画像を保存することで1つのツイートに対して複数の画像を保存できるようなテーブル構成になると思います。

> tweetですね。。変更しました。

はい！そうです！ありがとうございます！

> imageカラムを新しく作りました。

いいですね。これで画像を保存するカラムができたと思います。あとはこのimageカラムの型を追加していただければと思います。またテーブル名が`tweet_pictures`に対してカラム名が`image` になっているので、テーブル名とカラム名の命名の辻褄が合うように修正した方がいいかなと思いました。

ご確認のほどよろしくお願いいたします。

**Comment by yohei04** [comment]
>imageカラムの型を追加
>テーブル名とカラム名の命名の辻褄が合うように修正

変更致しました。
こちらで問題ないでしょうか？

<img width="605" alt="スクリーンショット 2020-02-06 14 18 10" src="https://user-images.githubusercontent.com/54460011/73908013-aa4e6400-48eb-11ea-8029-460b28d58924.png">

**Comment by daijiromaeyama** [comment]
修正ありがとうござます！問題ないです！
LGTMです👍

テーブル設計お疲れさまでした！
